### PR TITLE
III-5544 polyfill mainimage

### PR DESCRIPTION
**Pull request by JonasVHG**
### Added

- Polyfill to remove `image` from `json` if no `mediaObjects` are present
- tests for new Polyfill

### Note
- This does not handle the edge case where an a legacy `mainImage` is removed but the `json` still contains other images.

---
Ticket: https://jira.uitdatabank.be/browse/III-5544

### CODE REVIEW

**Review by simon-debruijn — APPROVED** [review]

File: src/Offer/ReadModel/JSONLD/PropertyPolyfillOfferRepository.php [hunk 1]
```diff
@@ -264,6 +265,20 @@ function (array $json) {
         );
     }
 
+    private function fixMainImage(JsonDocument $jsonDocument): JsonDocument
```
Comment (LucWollants) [inline src/Offer/ReadModel/JSONLD/PropertyPolyfillOfferRepository.php:268]: I would make the name of the method a bit more explicit. I think the names of the test methods describe very well what happens. For example `removeMainImageWhenMediaObjectIsEmpty`

**Review by LucWollants — APPROVED** [review]
